addOns/accessControl/src/main/java/org/zaproxy/zap/extension/accessControl/AccessControlAPI.java

```diff
@@ -0,0 +1,242 @@
+                        new String[] {PARAM_CONTEXT_ID, PARAM_USER_ID},
+                        new String[] {
+                            PARAM_UNAUTH_USER, PARAM_RAISE_ALERT, PARAM_ALERT_RISK_LEVEL
+                        }));
+
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_WRITE_HTML_REPORT, new String[] {PARAM_CONTEXT_ID, PARAM_FILENAME}));
+
+        this.addApiView(new ApiView(VIEW_GET_SCAN_PROGRESS, new String[] {PARAM_CONTEXT_ID}));
+        this.addApiView(new ApiView(VIEW_GET_SCAN_STATUS, new String[] {PARAM_CONTEXT_ID}));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        ApiResponse result = null;
+
+        switch (name) {
+            case ACTION_SCAN:
+                LOGGER.debug("Access control start scan called");
+
+                AccessControlScanStartOptions startOptions = new AccessControlScanStartOptions();
+
+                startOptions.targetContext = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID);
+
+                Mode mode = Control.getSingleton().getMode();
+                if (Mode.safe.equals(mode)) {
+                    throw new ApiException(
+                            ApiException.Type.MODE_VIOLATION,
+                            Constant.messages.getString(
+                                    "accessControl.scanOptions.error.mode.safe"));
+                } else if (Mode.protect.equals(mode)) {
+                    if (!startOptions.targetContext.isInScope()) {
+                        throw new ApiException(
+                                ApiException.Type.MODE_VIOLATION,
+                                Constant.messages.getString(
+                                        "accessControl.scanOptions.error.mode.protected",
+                                        startOptions.targetContext.getName()));
+                    }
+                }
+
+                if (usersExtension == null) {
+                    usersExtension =
+                            Control.getSingleton()
+                                    .getExtensionLoader()
+                                    .getExtension(ExtensionUserManagement.class);
+                }
+
+                List<User> users = new ArrayList<User>();
+
+                String[] commaSeparatedUserIDs =
+                        ApiUtils.getNonEmptyStringParam(params, PARAM_USER_ID).split("\\s*,\\s*");
+
+                for (int i = 0; i < commaSeparatedUserIDs.length; i++) {
+                    int userID;
+                    try {
+                        userID = Integer.parseInt(commaSeparatedUserIDs[i]);
+                    } catch (NumberFormatException nfe) {
+                        throw new ApiException(
+                                ApiException.Type.BAD_FORMAT, "Failed to parse userID (int).", nfe);
```

Comment:
We should use other type (`ILLEGAL_PARAMETER`?), the `BAD_FORMAT` is for when the requested format (e.g. xml, json) is invalid.